lib/open_ai_client.rb

```diff
@@ -0,0 +1,31 @@
+require "prompt_body"
+
+class OpenAiClient
+  OPEN_AI_ENDPOINT = "https://api.openai.com/v1/chat/completions"
+
+  def initialize(http_client, api_key)
+    @http_client = http_client
+    @api_key = api_key
+  end
+
+  def prompt(prompt)
+    @http_client.post(
+      OPEN_AI_ENDPOINT,
```

Comment:
not sure if this is really what you were asking about re keyword args, but if there is an option to use keyword args somewhere, I'd use them here because e.g. what is the arg we are sending `nil` for?

Reply:
Yes, this is the example I considered when discussing keyword arguments. The `nil` is a placeholder for a later change. I can remove it if it would make sense now. My plan is that it will take a class that will parse the OpenAI response. What do you think?

Reply:
I think I need to read more about nullables before commenting further :)